### ISSUE TITLE
Design needed for "tool tips"

### ISSUE THREAD

**Issue by jesicarson** [participedia/frontend#624]
Andrea to work on research of precedents and come up with a look and feel for ppedia to review with Jesi. Could be a full click through tutorial, could be hover on certain elements to reveal tips, etc. 

J&A to come up with a list of what elements should have tool tips (and text needed for them) here:

- [ ] Quick submit button: "Add a case, method or organization"

- [ ] Edit button: "Edit or add data to this case/ method/ organization"

- [ ] Search: "For advanced search options see [link to help faq about search syntax]"

etc...

**Comment by andreadelrio** [comment]
I've found some alternatives for this: tooltips (that are less intrusive) and feature tours (that take more of a centre stage e.g. lightbox).

Tooltips example
![image](https://user-images.githubusercontent.com/4016496/30833092-6d531e5c-a202-11e7-8136-77d393183662.png)

Feature tour - lightbox
![image](https://user-images.githubusercontent.com/4016496/30833123-86eba7c6-a202-11e7-88b9-35f2e0eb6ad4.png)

Hybrid: tooltips with multiple steps (slider)
![image](https://user-images.githubusercontent.com/4016496/30833206-cb01c3aa-a202-11e7-897f-1330fcaa4f96.png)

Some options to implement on the tech side:
http://kushagragour.in/lab/hint/
http://introjs.com/
http://bootstraptour.com/
https://gilbarbara.github.io/react-joyride/

**Comment by jesicarson** [comment]
Cool! Can you post links to those examples so I can go through the experiences live? On first glance the hybrid option seems to make sense. I'm curious what triggers the tool tips to pop up in option 1.

**Comment by andreadelrio** [comment]
For the first two I only have screenshots that I took from this [article]
(https://www.appcues.com/blog/the-5-best-walkthrough-examples)
First site is https://www.getflow.com/
The other one is https://inbox.google.com/?pli=1.
I can create an account in Flow to see how their tooltips work.

For the hybrid option you can look here:
http://introjs.com/ (click 'Demo')
https://gilbarbara.github.io/react-joyride/

**Comment by jesicarson** [comment]
Andrea and I like react joyride. She's going to research implementation and mockup the quick submit button using this text:

"Clicking quick submit allows you to publish a case, method or organization in minutes, or you can also submit a full entry here." 

Order of the tour would likely be 
1. search
2. sort results by cases, methods, orgs
3. quick submit

**Comment by andreadelrio** [comment]
![desktop hd copy](https://user-images.githubusercontent.com/4016496/31265413-908f5e2a-aa23-11e7-9214-f10c08791363.png)
![desktop hd](https://user-images.githubusercontent.com/4016496/31265414-909423b0-aa23-11e7-9c78-943d5c9aeed0.png)
@jesicarson  Here are some options, what do you think?

**Comment by jesicarson** [comment]
Nice! I like the top one better without the extra red line. You decided to do the dark background, or is that part of the built in joyride package?

**Comment by andreadelrio** [comment]
@jesicarson It's part of Joyride and can be disabled if needed.

**Comment by jesicarson** [comment]
Well, since your initial thought was to not have the dark background, perhaps try a quick mock of that and we can compare. Would also be good to see the little notifyer thingy, or show how you envision users being prompted to take the tour. Also if it's not a first time visitor, can we make sure the tour doesn't interrupt their session? And, we should consider where a user can go to find the tour again once it's been closed (maybe in the help side bar?)

**Comment by andreadelrio** [comment]
Here's the version with no dark overlay. 
![desktop_no_light](https://user-images.githubusercontent.com/4016496/31329240-0156b6f0-ac8e-11e7-9cee-0d89327a0399.png)

However, when you see the demo live, having a dark overlay really adds to the tour (it's like the camera pans to the spot we want to highlight), I can show you in our next meeting.

The notifier is this beacon which is animated.
<img width="1440" alt="beacon" src="https://user-images.githubusercontent.com/4016496/31329250-0b15afa2-ac8e-11e7-8511-aa2a02239ce8.png">


I need to check how we are handling users to see how we would disable this. And also figure out how we would trigger the tour at the user's request. I like the help sidebar idea.

**Comment by jesicarson** [comment]
I agree, the dark is much more clear and directs you to the feature more clearly. I like it! 
Very important to figure out the interaction with return users. Looking forward to seeing this! 

I'll keep working on some text for the rest of the features in the [copy deck](https://docs.google.com/document/d/1eUVXSBHN_QqR5O_MCNHpiplAAr51K3EM5yOo8w05I5k/edit?usp=sharing) under the help section, and I'll let you know here when it's ready to go.

**Comment by jesicarson** [comment]
@andreadelrio does the feature tour allow you to open other pages during the tour? ie: clicking quick sumbit, would we be able to show people the actual page that comes up, and then what the form looks like? Another example would be the reader view of a case - if they're doing the tour from the homepage, we want them to know how to edit existing content, but how do they get there?

**Comment by jesicarson** [comment]
There's some new text for the feature tour in the copy deck that will hopefully let you see how I got to that last question. Thanks!

**Comment by jesicarson** [comment]
Just for reference:

![screen shot 2017-10-26 at 1 12 19 pm](https://user-images.githubusercontent.com/13560530/32146509-e61d6844-bc95-11e7-88d2-5c2240c6c654.png)

![screen shot 2017-10-26 at 1 12 38 pm](https://user-images.githubusercontent.com/13560530/32146510-e631f78c-bc95-11e7-9737-35409e2bc80b.png)

![screen shot 2017-10-26 at 1 13 02 pm](https://user-images.githubusercontent.com/13560530/32146511-e64ceb28-bc95-11e7-883c-0316b7475c3a.png)

![screen shot 2017-10-26 at 1 13 17 pm](https://user-images.githubusercontent.com/13560530/32146512-e661719c-bc95-11e7-8e7d-84fb9fe0bc25.png)

![screen shot 2017-10-26 at 1 13 26 pm](https://user-images.githubusercontent.com/13560530/32146513-e6754db6-bc95-11e7-8e13-4384658d2c4f.png)

**Comment by jesicarson** [comment]
@andreadelrio 

Notes from Nov 20 D&T meeting:

- added a few new tips
- changes in copy deck highlighted yellow

**Comment by jesicarson** [comment]
@andreadelrio is going to check if the start tour button in the help menu can be triggered based on the page you are on.

**Comment by jesicarson** [comment]
@andreadelrio is going to make an invisible element to target the feedback button (usersnap) as a work around.

**Comment by jesicarson** [comment]
Update @andreadelrio :
- the show tour button will redirect to the homepage unless the user is on an edit form page, in which case the "show tour" wouldn't be visible on the help sidebar (because it would navigate away from an edit form and could lose unpublished data)
- Remove internal tours (ie on case reader or edit forms) * but we still need to design the info icon system in the form, either a hover system or question mark icon.

**Comment by jesicarson** [comment]
User Feedback: I'd be in favour of a single popup after 1-2 minutes spent on the site with a really short tagline, button to login/signup (with social media signin/up options), and a newsletter tick-box.

**Comment by jesicarson** [comment]
example (this popped up after about 30 seconds of visiting the everyday democracy site):
![screen shot 2018-12-20 at 11 56 22 am](https://user-images.githubusercontent.com/13560530/50307845-57e38700-044e-11e9-8c94-b0d816cd726c.png)

**Comment by jesicarson** [comment]
@ascott Here's the old issue for the feature tour that Andrea built. As noted above, text for the tour was drafted in the copy deck, but i'll make a new doc. let me know if you want to keep using this issue or make a new one in API for this. 

I was able to pull some screenshots from the old testing site a while back too, so here's where we were at with the visual design:

This screenshot shows the red beacon, which had a sort of pulse animation on it, and would come up for new visitors only: 
![Screen Shot 2018-11-26 at 10 50 11 AM](https://user-images.githubusercontent.com/13560530/62395285-984acd80-b524-11e9-82bc-a41cfcdc3766.png)

This shows the dark background and modal popup with arrow pointing to the feature that was identified by the beacon (not sure the arrow will work with all features, placement wise, though):
![Screen Shot 2018-11-26 at 10 50 25 AM](https://user-images.githubusercontent.com/13560530/62395287-9b45be00-b524-11e9-86b9-e6246b03c618.png)

This shows how we envisioned users bringing back the tour if they had closed it, finding it in the help & contact menu:
![Screen Shot 2018-11-26 at 10 51 38 AM](https://user-images.githubusercontent.com/13560530/62395293-9e40ae80-b524-11e9-8890-b18b331892ac.png)

**Comment by jesicarson** [comment]
Saw this prompt to do a tutorial on Github today, we could think about specific tutorials, push to contribution guidelines, teaching materials, collections, etc... 
![Screen Shot 2019-08-09 at 11 04 13 AM](https://user-images.githubusercontent.com/13560530/62801333-5ed10f80-ba9a-11e9-8852-c7ed28672af2.png)

**Comment by jesicarson** [comment]
also this: https://introjs.com/
